mne/datasets/utils.py

```diff
@@ -426,32 +259,97 @@ def _data_path(path=None, force_update=False, update_path=True, download=True,
             if answer.lower() != 'y':
                 raise RuntimeError('You must agree to the license to use this '
                                    'dataset')
-        assert len(url) == len(hash_)
-        assert len(url) == len(archive_name)
-        assert len(url) == len(folder_orig)
-        assert len(url) == len(folder_path)
-        assert len(url) > 0
-        # 1. Get all the archives
-        full_name = list()
-        for u, an, h, fo in zip(url, archive_name, hash_, folder_orig):
-            remove_archive, full = _download(path, u, an, h)
-            full_name.append(full)
-        del archive_name
-        # 2. Extract all of the files
-        remove_dir = True
-        for u, fp, an, h, fo in zip(url, folder_path, full_name, hash_,
-                                    folder_orig):
-            _extract(path, name, fp, an, fo, remove_dir)
-            remove_dir = False  # only do on first iteration
-        # 3. Remove all of the archives
-        if remove_archive:
-            for an in full_name:
-                os.remove(op.join(path, an))
-
-        logger.info('Successfully extracted to: %s' % folder_path)
-
-    _do_path_update(path, update_path, key, name)
-    path = folder_path[0]
+
+    # downloader & processors TODO: may want to skip using tqdm during tests?
+    progressbar = True
+    if name == 'fake':
+        progressbar = False
+    downloader = pooch.HTTPDownloader(progressbar=progressbar)  # use tqdm
+    unzip = pooch.Unzip(extract_dir=path)  # to unzip downloaded file
+    untar = pooch.Untar(extract_dir=path)  # to untar downloaded file
+
+    # this processor handles nested tar files
+    nested_untar = pooch.Untar(extract_dir=op.join(path, FOLDER_NAMES[name]))
+
+    # create a map for each dataset name to its corresponding processor
+    # Note: when adding a new dataset, a new line must be added here.
+    processors = dict(
+        bst_auditory=nested_untar,
+        bst_phantom_ctf=nested_untar,
+        bst_phantom_elekta=nested_untar,
+        bst_raw=nested_untar,
+        bst_resting=nested_untar,
+        fake=untar,
+        fieldtrip_cmc=nested_untar,
+        kiloword=untar,
+        misc=untar,
+        mtrf=unzip,
+        multimodal=untar,
+        fnirs_motor=untar,
+        opm=untar,
+        sample=untar,
+        somato=untar,
+        spm=untar,
+        testing=untar,
+        visual_92_categories=untar,
+        phantom_4dbti=unzip,
+        refmeg_noise=unzip,
+        hf_sef_raw=pooch.Untar(
+            extract_dir=path, members=[f'hf_sef/{subdir}' for subdir in
+                                       ('MEG', 'SSS', 'subjects')]),
+        hf_sef_evoked=pooch.Untar(
+            extract_dir=path, members=[f'hf_sef/{subdir}' for subdir in
+                                       ('MEG', 'SSS', 'subjects')]),
```

Comment:
@drammock do you know why we needed hf_Sef to be like this?  These LOC I copied seems to break  circle CI in : https://github.com/mne-tools/mne-python/issues/9774

It seems to work perfectly fine w/ `processor = 'untar'` untaring to the path. I confirm that the error was not raised when I changed the processor to just regular `pooch.Untar(extract_dir=path`

Reply:
To clarify, I copied these from your original PR aimed to tackle #8674